lib/mayaUsd/render/MaterialXGenOgsXml/ShaderGenUtil.cpp

```diff
@@ -138,7 +147,16 @@ void TopoNeutralGraph::computeGraph(const mx::ElementPtr& material)
             if (!sourceInput) {
                 continue;
             }
-            auto              connectedNode = sourceInput->getConnectedNode();
+            auto connectedNode = sourceInput->getConnectedNode();
+
+            // In textured mode we traverse everything, but in untextured mode we only traverse PBR
+            // level connections.
+            static const auto isPbrType
+                = [](const auto& t) { return t == "BSDF" || t == "EDF" || t == "surfaceshader"; };
```

Comment:
[Nitpick] Can we compare against the string constants from `MaterialXCore/Types.h` instead? E.g.: `t == mx::BSDF_TYPE_STRING`.

[Question] Do we need to worry about volume-, light- and displacement shaders at all?

Reply:
This code already leans heavily towards processing surface materials. I suspect it will never hit a lightshader connection.